HW06.Rmd

```diff
@@ -1,92 +1,181 @@
+
+```{r load in data, message=FALSE}
+library(tidyverse) 
+library(dplyr) 
+library(knitr)
+
+#First, get the list of all file names (and then paste "./RNA_Seq_processed" to get the full path)
+file_names_WT <- list.files("./RNA_Seq_processed/WT", full.names=TRUE)
+file_names_H3K27M <- list.files("./RNA_Seq_processed/H3K27M", full.names=TRUE)
+
+#Read in all files! After reading files, use function select within an apply to get tables with the columns of interest. Then, change the TPM column name to include info on the WT/H3K27M status
+
+#I realize creating a function might have been much better, but I could not get them to work! Therefore, I decided to use for loops and apply functions which I have used more often in the past :)
+
+# WT
+files_WT_full <- lapply(file_names_WT, read_tsv)
+files_WT <- lapply(files_WT_full, select,contains(c("Name","TPM")))
+for (i in 1: length(files_WT))
+{
+  colnames(files_WT[[i]]) <- c("gene_id",paste0("TPM_",i,"_WT")) 
+}
+
+# H3K27M
+files_H3K27M_full <- lapply(file_names_H3K27M, read_tsv)
+files_H3K27M <- lapply(files_H3K27M_full, select,contains(c("Name","TPM")))
+for (i in 1: length(files_H3K27M))
+{
+  colnames(files_H3K27M[[i]]) <- c("gene_id",paste0("TPM_",i,"_H3K27M")) 
+}
+
+#Get a list with all files!
+files <- c(files_WT,files_H3K27M)
+
+#Merge all the files in one table. I used a for loop and the function "full join" to acomplish this.
+GBM.transcripts <- files[[1]]
+for (i in 2: length(files))
+{
+  GBM.transcripts <- full_join(GBM.transcripts,files[[i]],by = "gene_id") 
+}
+```
+
+Now, install the packages commented below (if needed), and then use this code to map the transcript IDs to gene symbols. To use this code, you need a dataframe called GBM.transcripts that has the first column "gene_id" that contains the transcript ids (e.g. ENST00000456328.2) and the remaining columns contain the TPM data. So long as the first column contains the "gene_id" column as mentioned above, this *should* run. 
+
+```{r map gene.ids to gene.symbols using ensembldb, message=FALSE,warning=FALSE}
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#NN
+#install.packages("XML", repos = "http://www.omegahat.net/R")
+library(XML)
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
```

Comment:
I would be eternally grateful if you could explain this code to me in person one day :,)
I am utter garbage at merging and finagling data.